trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/credentials/SigningController.java

```diff
@@ -146,7 +177,10 @@ private boolean isValidAuthorization(
             String httpMethod,
             Optional<byte[]> entity)
     {
-        String expectedAuthorization = signRequest(metadata, credentialsSupplier, requestURI, requestHeaders, queryParameters, httpMethod, entity);
-        return authorizationHeader.equals(expectedAuthorization);
+        // temp workaround until https://github.com/airlift/airlift/pull/1178 is accepted
+        return Stream.of(Mode.values()).anyMatch(mode -> {
```

Comment:
How come we validate headers either in their default representation or lowercasing them? I'd assume only one of the two should work?

Reply:
This is a bug (bad feature) in Jetty. Jetty is caching header values (I'm not sure why) and their cache is _not_ case sensitive. In this case, a version of `content-type` will get cached with the charset in the wrong case and this means we can no longer verify the signature. I have a PR in Airlift to make Jetty's cache case sensitive (they have a flag for this).

Reply:
Interesting... thanks for the info!